core/chains/evm/client/node_lifecycle_test.go

```diff
@@ -127,15 +140,22 @@ func TestUnit_NodeLifecycle_aliveLoop(t *testing.T) {
 		threshold := 3
 		cfg := TestNodeConfig{PollFailureThreshold: uint32(threshold), PollInterval: testutils.TestInterval}
 		var calls atomic.Int32
-		n := newTestNodeWithCallback(t, cfg, func(method string, params gjson.Result) (respResult string, notifyResult string) {
+		n := newTestNodeWithCallback(t, cfg, func(method string, params gjson.Result) (resp testutils.JSONRPCResponse) {
 			switch method {
+			case "eth_subscribe":
+				resp.Result = `"0x00"`
+				resp.Notify = HeadResult
+				return
+			case "eth_unsubscribe":
+				return
 			case "web3_clientVersion":
 				defer calls.Inc()
-				return "this will error", ""
```

Comment:
This test was problematic and the root cause for introducing a more complex return type `JSONRPCResponse`. The error triggered by this return value was actually from malformed JSON, which was sufficient before when there was no active subscription, but now the side-effect of unsubscribing causes cascading effects that break the test. Returning a valid jsonrpc error instead restores the original behavior.